### PR TITLE
feat: update map design doc

### PR DESCRIPTION
**Pull request by kminoda**
## Description

Update map design doc. Related to: https://github.com/autowarefoundation/autoware.universe/pull/3986

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/design/autoware-architecture/map/index.md [hunk 1]
```diff
@@ -63,3 +89,14 @@ The vector cloud map must be supplied as a file with the following requirements:
...
+### Projection information
+
+The projection information must be supplied as a file with the following requirements:
+
+- It must be in YAML format, provided into `map_projection_loader` in current Autoware Universe implementation.
+- The file must contain the following information:
+  - The name of the projection method used to convert between local and global coordinates
+  - The parameters of the projection method (depending on the projection method)
+
+For further information, please refer to [the Readme of `map_projection_loader` in Autoware Universe](https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_loader/README.md).
```
Comment (YamatoAndo) [inline docs/design/autoware-architecture/map/index.md:102]:
```suggestion
For further information, please refer to [the readme of `map_projection_loader` in Autoware Universe](https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_loader/README.md).
```

File: docs/design/autoware-architecture/map/index.md [hunk 2]
```diff
@@ -63,3 +89,14 @@ The vector cloud map must be supplied as a file with the following requirements:
 !!! warning
 
     Under Construction
+
+### Projection information
```
Comment (mitsudome-r) [inline docs/design/autoware-architecture/map/index.md:93]:
```suggestion
### Projection Information
```

File: docs/design/autoware-architecture/map/index.md (hunk 1 above)
Comment (mitsudome-r) [reply docs/design/autoware-architecture/map/index.md:102]: Also, I think the link should be 
`https://github.com/autowarefoundation/autoware.universe/blob/main/map/map_projection_loader/README.md`

File: docs/design/autoware-architecture/map/index.md (hunk 2 above)
Comment (kminoda) [reply docs/design/autoware-architecture/map/index.md]: Fixed: https://github.com/autowarefoundation/autoware-documentation/pull/431/commits/e1920074b317749ccd901d9175caa159a5794526

File: docs/design/autoware-architecture/map/index.md (hunk 1 above)
Comment (kminoda) [reply docs/design/autoware-architecture/map/index.md]: Fixed: https://github.com/autowarefoundation/autoware-documentation/pull/431/commits/ac37caaed7c70a631f7ccf735464a149ea55b807

File: docs/design/autoware-architecture/map/index.md [hunk 3]
```diff
@@ -17,24 +17,51 @@ A vector map contains highly accurate information about a road network, lane geo
...
+- **To API layer**
+  - Projection information: Used to convert localization results from local coordinate system to geodetic coordinate system
 
 ## 5. Map Specification
 
 ### Point Cloud Map
 
 The point cloud map must be supplied as a file with the following requirements:
 
+- The pointcloud map should be projected on the same coordinate defined in `map_projection_loader` in order to be consistent with the lanelet2 map and other packages that converts between local and geodetic coordinates. For more information, please refer to [the readme of `map_projection_loader`](https://github.com/autowarefoundation/autoware.universe/tree/main/map/map_projection_loader/README.md).
```
Comment (YamatoAndo) [inline docs/design/autoware-architecture/map/index.md:64]:
```suggestion
- The point cloud map should be projected on the same coordinate defined in `map_projection_loader` in order to be consistent with the lanelet2 map and other packages that converts between local and geodetic coordinates. For more information, please refer to [the readme of `map_projection_loader`](https://github.com/autowarefoundation/autoware.universe/tree/main/map/map_projection_loader/README.md).
```

Comment (kminoda) [reply docs/design/autoware-architecture/map/index.md]: Fixed! https://github.com/autowarefoundation/autoware-documentation/pull/431/commits/352a8db641c285409e503e713f55a53600a61493

**Review by YamatoAndo — APPROVED** [review]

**Pull request merged**